dali/core/cuda_stream_pool_test.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <cuda_runtime.h>
+#include <random>
+#include <thread>
+#include <vector>
+#include "dali/core/cuda_error.h"
+#include "dali/core/cuda_stream_pool.h"
+#include "dali/core/cuda_stream.h"
+
+namespace dali {
+
+class CUDAStreamPoolTest : public ::testing::Test {
```

Comment:
why isn't this class inside the test namespace?

Reply:
Because I didn't want to put this namespace in the proper header. IMO it look weird to have
```C++
namespace test {
class CUDAStreamPoolTest;
}```
in `cuda_stream_pool.h`